### PR TITLE
datamodel: use equals for string comparison.

### PR DESCRIPTION
**Pull request by ardevd**
It is almost always a mistake to compare two instances of `java.lang.String` using reference equality == or !=, because it is not comparing the actual value but rather locations in memory.

### CODE REVIEW

File: src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java [hunk 1]
```diff
@@ -1471,7 +1471,7 @@ public void saveSharesDB(List<OCShare> shares) {
         // Reset flags & Remove shares for this files
         String filePath = "";
         for (OCShare share : shares) {
-            if (filePath != share.getPath()) {
+            if (filePath.equals(share.getPath())) {
```
Comment (ezaquarii) [inline src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java]: You've changed `!=` to `equals`. Are you sure the logic is good here?

Comment (ardevd) [reply src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java]: *Embarrassed*. Thank you. Nice catch. Fixed.

Comment (ezaquarii) [reply src/main/java/com/owncloud/android/datamodel/FileDataStorageManager.java]: And how do you test this code path?

**Comment by AndyScherzinger** [comment]
@tobiasKaminsky to follwo-up on @ezaquarii's question: How can we get this to tested via a unit test? :)

**Comment by ardevd** [comment]
@AndyScherzinger I can do some Powermock magic and write a unit test for this I think.

**Comment by AndyScherzinger** [comment]
> @AndyScherzinger I can do some Powermock magic and write a unit test for this I think.

What do you think @tobiasKaminsky @ezaquarii? Powermock would be fine with me.

**Comment by tobiasKaminsky** [comment]
Not sure if powermock is needed, as we already have mockito. 
So I would first try it with mockito. 

To test this function the only (maybe) tricky part is to catch applyBatch(…) but this should be possible with pure Mockito.

**Comment by tobiasKaminsky** [comment]
Ping me, if you need any help :+1:

**Pull request merged**